modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/consensus/ConsensusManager.scala

```diff
@@ -0,0 +1,107 @@
+package org.tessellation.sdk.infrastructure.consensus
+
+import cats.effect.Async
+import cats.kernel.Next
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+import cats.syntax.show._
+import cats.syntax.traverse._
+import cats.{Applicative, Order, Show}
+
+import scala.reflect.ClassTag
+import scala.reflect.runtime.universe.TypeTag
+
+import org.tessellation.ext.cats.syntax.next._
+import org.tessellation.sdk.domain.consensus.ConsensusFunctions
+
+import org.typelevel.log4cats.slf4j.Slf4jLogger
+
+trait ConsensusManager[F[_], Event, Key, Artifact] {
+
+  def checkForTrigger(event: Event): F[Unit]
+  def checkForStateUpdate(key: Key)(resources: ConsensusResources[Artifact]): F[Unit]
+}
+
+object ConsensusManager {
+
+  def make[F[_]: Async, Event, Key: Show: Order: Next: TypeTag: ClassTag, Artifact <: AnyRef: Show: TypeTag](
+    consensusFns: ConsensusFunctions[F, Event, Key, Artifact],
+    consensusStorage: ConsensusStorage[F, Event, Key, Artifact],
+    consensusStateUpdater: ConsensusStateUpdater[F, Key, Artifact]
+  ): ConsensusManager[F, Event, Key, Artifact] =
+    new ConsensusManager[F, Event, Key, Artifact] {
+
+      private val logger = Slf4jLogger.getLoggerFromClass[F](ConsensusManager.getClass)
+
+      def checkForTrigger(event: Event): F[Unit] =
+        consensusStorage.getLastKeyAndArtifact.flatMap(_.traverse(internalForCheckTrigger(_, event)).void)
+
+      def checkForStateUpdate(key: Key)(resources: ConsensusResources[Artifact]): F[Unit] =
+        consensusStorage.getState(key).flatMap(_.traverse(internalCheckForStateUpdate(key, _, resources)).void)
+
+      private def internalForCheckTrigger(lastKeyAndArtifact: (Key, Artifact), event: Event): F[Unit] =
```

Comment:
shouldn't it be `internalCheckForTrigger`?

Reply:
Yes, absolutely